python/ray/_private/gcs_pubsub.py

```diff
@@ -112,21 +192,19 @@ def __init__(
             address: str = None,
             channel: grpc.Channel = None,
     ):
+        super().__init__()
+
         if address:
             assert channel is None, \
                 "address and channel cannot both be specified"
             channel = gcs_utils.create_gcs_channel(address)
         else:
             assert channel is not None, \
                 "One of address and channel must be specified"
-        self._lock = threading.RLock()
+        self._lock = threading.Lock()
+        self._cond = threading.Condition(lock=self._lock)
```

Comment:
It'll also be useful to add some inline comments about what are these protecting